### PR TITLE
Separate the function to record a new error.

### PR DESCRIPTION
**Pull request by ariya**
This is extracted from PR #262.

Refs #1042.

### CODE REVIEW

**Comment by ariya** [comment]
@scottgonzalez @mikesherov Something is wrong the CLA checker?

**Comment by scottgonzalez** [comment]
Looks like you sent this PR before you signed the CLA. The automated check does not rerun on new signatures (see https://github.com/jquery/jquery-license/issues/6). I've rerun the check and it marked the commit as valid.

**Comment by ariya** [comment]
Thanks a lot @scottgonzalez!

File: esprima.js [hunk 1]
```diff
@@ -1992,6 +1992,22 @@
...
 
+
+    function recordError(error) {
+        var e, existing;
+
+        for (e = 0; e < extra.errors.length; e++) {
+            existing = extra.errors[e];
+            // Prevent duplicated error.
+            /* istanbul ignore next */
+            if (existing.index === error.index && existing.message === error.message) {
```
Comment (ikarienator) [inline esprima.js]: This is going to be slow...

Comment (ariya) [inline esprima.js]: Only for tolerant mode, which is expected not to be superfast anyway (for now).

**Comment by ikarienator** [comment]
:+1:

**Comment by mikesherov** [comment]
:+1:

**Comment by mikesherov** [comment]
Perhaps we can do this for comment attachment next!

**Pull request closed without merging**